src/System.Linq.Expressions/tests/Conditional/ConditionalTests.cs

```diff
@@ -0,0 +1,237 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests
+{
+    public static class ConditionalTests
+    {
+        private class Truthiness
+        {
+            private bool Value { get; }
+
+            public Truthiness(bool value)
+            {
+                Value = value;
+            }
+
+            public static implicit operator bool(Truthiness truth) => truth.Value;
+
+            public static bool operator true(Truthiness truth) => truth.Value;
+
+            public static bool operator false(Truthiness truth) => !truth.Value;
+
+            public static Truthiness operator !(Truthiness truth) => new Truthiness(!truth.Value);
+        }
```

Comment:
Nit: can we move these helpers to the bottom, rather than interspersing them with facts?